### PR TITLE
Add useReviews factory with tests

### LINKED ISSUE #4775: useReview factory
Create a Next factory function and typing for `useReview` composable with the following interface:

```js
const { reviews, search, loading, error } = useReview()
```

TODO
- types in `core` package
- factory in `core` package
- tests

### PR DESCRIPTION
**Pull request by filipsobol**
Adds `useReviewsFactory` with types and tests, closes #4775.

Additionally, I added:
* `addReview` to submit new review,
* `totalReviews` and `averageRating` properties in case there are more reviews than we're able to fetch at once (pagination).

### CODE REVIEW

File: packages/core/core/src/factories/useReviewsFactory.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        totalReviews.value = total;
+        averageRating.value = average;
+      } catch (searchError) {
+        error.value = searchError.toString();
+      } finally {
+        loading.value = false;
+      }
+    };
+
+    const add = async (params: REVIEW_ADD_PARAMS): Promise<void> => {
```
Comment (andrzejewsky) [inline packages/core/core/src/factories/useReviewsFactory.ts:40]: addReview

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

**Review by filrak — APPROVED** [review]

**Comment by filrak** [comment]
Please add to changelog ;)

**Comment by filipsobol** [comment]
> Please add to changelog ;)

Done

**Review by andrzejewsky — APPROVED** [review]
@filipsobol missing interface for getters, but please do it in the next task

**Pull request merged**